### ISSUE TITLE
HTTP/3 Transport and Alt-Svc

### ISSUE THREAD

**Issue by LPardue** [w3c/webtransport#78]
During the WEBTRANS BoF at IETF Singapore 106 I made a comment at the microphone about the relationship between WebTransport over HTTP/3 and Alt-Svc. I was asked to open an issue here.

__Simple Problem__
The following are requirements from https://tools.ietf.org/html/draft-vvv-webtransport-http3-01#section-4.1:

*  In order to indicate support for Http3Transport, both the client and
   server MUST send an empty "http3_transport_support" transport
   parameter.  Endpoints MUST NOT use any Http3Transport-related
   functionality unless the parameter has been negotiated.

*  If "http3_transport_support" is negotiated, support for the QUIC
   DATAGRAM extension MUST be negotiated.

For any given HTTP/3 connection with a value of "h3-draft" that meets the following above criteria, what happens if an Alt-Svc advert on that connection incorrectly causes a client to attempt a new connection that cannot meet the same requirements? I think marking the alternative as broken is probably good enough.

__Broader problem__
An HTTP/3 connection that coalesces connections might trip up if some of the traffic to origin is intended to be Http3Transport and some is not. An alternative for a non-Http3Transport-using authority might cause a complete changeover to fail. 

HTTP/2 defines the 421 status code that can be used to mitigate such problems but it doesn't apply to all possible error cases (only the one where extended CONNECT fails).

I think in reality, an Http3Transport server will likely want to be smart about the Alt-Svc values it produces. 

__Broader broader problem__
How does a client discover an Http3Transport service?

We're also talking about putting Alt-Svc in the DNS. How does a client learn of the availability of Http3Transport providing server based on its name, should it even be able to?

**Comment by LPardue** [comment]
@vasilvv I wonder if this question is better suited to IETF, any thoughts?

**Comment by vasilvv** [comment]
Most likely.  Once we adopt transport drafts, I'd have to go through issues here and see which of those are better suited to IETF; I suspect there's a bunch.

**Comment by wilaw** [comment]
@vasilvv  - is this an IETF issue? If so, can it be transferred and closed out here?

**Comment by vasilvv** [comment]
It's actually not entirely clear to me where Alt-Svc should live (in webtrans draft, somewhere in Fetch, in this draft, or where).  @martinthomson might have opinions on this.

**Comment by LPardue** [comment]
I think some of these questions have changed or been addressed as the specs have progressed. 

I wrote an I-D about a related problem in https://tools.ietf.org/html/draft-pardue-alt-svc-ext-unsupported-00.

It feels to me, now that the IETF has a WG and adopted HTTP/3 transport that the conversation can move there. However, I am not a browser implementer and I can't speak to whether any of the points I made in the OP spin out to concerns specific to this API. Do happy to here opinions.
